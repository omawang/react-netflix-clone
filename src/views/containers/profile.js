import React from 'react';

import { Header, Profiles } from '../components';
import * as ROUTES from '../../constants/routes';
import logo from '../../logo.svg';

export function SelectProfileContainer({ user, handleSelectProfile }) {
  return (
    <>
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src={logo} to={ROUTES.HOME} alt="Netflix" />
        </Header.Frame>
      </Header>

      <Profiles>
        <Profiles.Title>Who&apos;s watching</Profiles.Title>
        <Profiles.List>
          <Profiles.Item
            data-testid="user-profile"
            onClick={() =>
              handleSelectProfile({
                displayName: user.displayName,
                photoURL: user.photoURL,
              })
            }
          >
            <Profiles.Picture src={user.photoURL} />
            <Profiles.Name>{user.displayName}</Profiles.Name>
          </Profiles.Item>
        </Profiles.List>
      </Profiles>
    </>
  );
}
