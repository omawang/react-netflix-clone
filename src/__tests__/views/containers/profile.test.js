import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { fireEvent, render } from '@testing-library/react';

import { SelectProfileContainer } from '../../../views/containers/profile';

describe('<SelectProfileContainer />', () => {
  it('renders <SelectProfileContainer /> with populated data', () => {
    const user = { displayName: 'Awank', photoURL: '2' };
    const handleSelectProfile = jest.fn();
    const { container, getByTestId } = render(
      <Router>
        <SelectProfileContainer
          user={user}
          handleSelectProfile={handleSelectProfile}
        />
      </Router>
    );

    fireEvent.click(getByTestId('user-profile'));
    expect(handleSelectProfile).toHaveBeenCalled();

    expect(container.firstChild).toMatchSnapshot();
  });
});
