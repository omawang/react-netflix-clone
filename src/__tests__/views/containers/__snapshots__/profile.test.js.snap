// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectProfileContainer /> renders <SelectProfileContainer /> with populated data 1`] = `
<div
  class="header__Container-sc-1v45rsj-1 eSeMKt"
>
  <a
    href="/"
  >
    <img
      alt="Netflix"
      class="header__Logo-sc-1v45rsj-6 kbpjKE"
      data-testid="header-bg"
      src="logo.svg"
    />
  </a>
</div>
`;
